openedx-unsupported/edx-analytics-pipeline
# Generate edx.analytics.task and parameter documentation with Sphinx

This PR adds a custom `sphinx.ext.autodoc` plugin which appends `luigi.Parameter` attribute names and description values to the documentation generated for each `luigi.Task` class.

**JIRA tickets**: Implements [OC-1391](https://tasks.opencraft.com/browse/OC-1391), [OLIVE-16](https://openedx.atlassian.net/browse/OLIVE-16), [OLIVE-15](https://openedx.atlassian.net/browse/OLIVE-15)

**Discussions**: Implementation decisions came out of a discovery task document, created by @e-kolpakov, and reviewed by @mulby and @bradenmacdonald.

**Dependencies**: Sphinx 1.3.5

**Screenshots**:
[Generated docs.zip](https://github.com/edx/edx-analytics-pipeline/files/151665/docs.zip) **Notes** The attached docs.zip are the PDFs generated by merging the code from this PR with the description changes for PR #208.  They're also subject to print CSS, which removes the lovely tables of contents sidebar.

![edx.analytics.pipeline home](https://cloud.githubusercontent.com/assets/7556571/13397520/3c599a78-df4a-11e5-94c0-1cae7c80e35b.png)
![workflow_entry_points](https://cloud.githubusercontent.com/assets/7556571/13407296/a3c424f8-df77-11e5-8367-726ada2ddc9d.png)
![supporting_tasks](https://cloud.githubusercontent.com/assets/7556571/13407302/a8e5b47e-df77-11e5-8934-a1d2f082ca27.png)
![task index](https://cloud.githubusercontent.com/assets/7556571/13397573/7e77dd16-df4a-11e5-8ea5-daac131d2714.png)
![module index](https://cloud.githubusercontent.com/assets/7556571/13397576/81761c44-df4a-11e5-8843-88e1c9f3ba1e.png)

**Sandbox URL**: TBD 

**Testing instructions**: 
1. go to edx-analytics-pipeline 
2. make install 
3. make docs
4. open docs/index.html

Or check the attached PDF.

**Author notes and concerns**:
1. The Analytics Tasks section in `sphinx_source/index.rst` is automatically generated from the edx.analytics.tasks listed in `setup.cfg`, so any tasks missing from that list will not be documented.

**Reviewers**
- [] Code: (@e-kolpakov)
- [] Product: (TBD)
- [] Platform: (TBD)
- [] UX: (TBD)
- [] TNL: (TBD)
- [] a11y: (TBD)
- [] doc/strings: (TBD)
